### PR TITLE
[addons] move rollback feature to the update dialog

### PR DESCRIPTION
**Pull request by tamland**
This fixes and unifies the rollback and update feature by listing all available version (local packages or remote) in one place, in the update dialog.

### CODE REVIEW

**Comment by stefansaraev** [comment]
jenkins build this please

**Comment by mkortstiege** [comment]
Could you change the dialog to highlight only the currently used one? btw, do we want the users to be able to replace currently used with currently used?

Other than that i really like it ;)

**Comment by tamland** [comment]
@mkortstiege As we don't track the origin there is no way to know which is the currently used one, only the version. I agree it's a bit weird to highlight all versions, but that's what can be done currently.

**Comment by mkortstiege** [comment]
OK.

File: xbmc/addons/AddonInstaller.cpp [hunk 1]
```diff
@@ -259,6 +259,8 @@ bool CAddonInstaller::InstallFromZip(const std::string &path)
   if (!g_passwordManager.CheckMenuLock(WINDOW_ADDON_BROWSER))
     return false;
 
+  CLog::Log(LOGDEBUG, "CAddonInstaller: installing from zip '%s'", path.c_str());
```
Comment (mkortstiege) [inline xbmc/addons/AddonInstaller.cpp]: Since we logging the file path you should use `CURL::GetRedacted(path).c_str()` here.

**Comment by tamland** [comment]
good to go then?
jenkins build this please

**Comment by mkortstiege** [comment]
Jep, merge away once build successfully.

**Comment by MartijnKaijser** [comment]
Or "none available" or something

**Comment by phil65** [comment]
I think in that case the SelectDialog shouldnt even open, just a notification with "none available" would be enough.

**Comment by phil65** [comment]
...or perhaps even disabling that button.

**Comment by Razzeee** [comment]
Disable that button in that case should be preferred

**Comment by tamland** [comment]
I dislike disabling it as it would make the info dialog lag again. it's an architectural problem. Now, the delay is at least contained to the update dialog.

**Comment by tamland** [comment]
This is really weird btw. Why would an empty list outright break navigation? if that's the case we probably have a lot more broken dialogs.

**Comment by MartijnKaijser** [comment]
we prevented broken navigation by always adding an item iirc

**Comment by ronie** [comment]
> This is really weird btw. Why would an empty list outright break navigation?

we can't focus a list if it contains no listitems.

skins have to define the id of the control that should be focused when a window/dialog opens.
in case of the select dialog, it's the id of the list.

if the list is empty, kodi can't focus it. as a result no control will be focused.
in that case, onleft/onright/onup/ondown navigations are undefined
and you're basically stuck.

**Comment by tamland** [comment]
Then the correct solution is to focus the close button. Possible to fix in skin?

**Comment by ronie** [comment]
nope.

**Comment by tamland** [comment]
see #8416 / #8417

**Pull request merged**